Mafl update script failing

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

Mafl

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/mafl.sh)"

### ⚙️ What settings are you using?

- [ ] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 📝 Provide a clear and concise description of the issue.

Update is failing.

### 🔄 Steps to reproduce the issue.

1. enter Mafl LXC shell
2. type `update`

### ❌ Paste the full error output (if available).

```
  ⏳   Updating Mafl to vanges](https://github.com/hywax/mafl/compare/v0.15.3...v0.15.4)

# (Patience)curl: (3) unmatched close brace/bracket in URL position 55:
https://github.com/hywax/mafl/archive/refs/tags/vanges](https://github.com/hywax/mafl/compare/v0.15.3...v0.15.4)\r\n\r\n#.tar.gz
                                                      ^

[ERROR] in line 33: exit code 0: while executing command curl -fsSL "https://github.com/hywax/mafl/archive/refs/tags/v${RELEASE}.tar.gz" -o $(basename "https://github.com/hywax/mafl/archive/refs/tags/v${RELEASE}.tar.gz")
```

### 🖼️ Additional context (optional).

_No response_

--- a/ct/mafl.sh
+++ b/ct/mafl.sh
@@ -30,12 +30,19 @@ function update_script() {
 
   RELEASE=$(curl -fsSL https://api.github.com/repos/hywax/mafl/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
   if [[ "${RELEASE}" != "$(cat ~/.mafl 2>/dev/null)" ]] || [[ ! -f ~/.mafl ]]; then
-    msg_info "Updating Mafl to v${RELEASE} (Patience)"
+    msg_info "Stopping Mafl service"
     systemctl stop mafl
+    msg_ok "Service stopped"
+
+    msg_info "Performing backup"
     mkdir -p /opt/mafl-backup/data
     mv /opt/mafl/data /opt/mafl-backup/data
     rm /opt/mafl
+    msg_ok "Backup complete"
+    
     fetch_and_deploy_gh_release "mafl" "hywax/mafl"
+
+    msg_info "Updating Mafl to v${RELEASE}"
     cd /opt/mafl
     yarn install
     yarn build

--- a/ct/mafl.sh
+++ b/ct/mafl.sh
@@ -27,18 +27,24 @@ function update_script() {
     msg_error "No ${APP} Installation Found!"
     exit
   fi
+
   RELEASE=$(curl -fsSL https://api.github.com/repos/hywax/mafl/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
-  msg_info "Updating Mafl to v${RELEASE} (Patience)"
-  systemctl stop mafl
-  curl -fsSL "https://github.com/hywax/mafl/archive/refs/tags/v${RELEASE}.tar.gz" -o $(basename "https://github.com/hywax/mafl/archive/refs/tags/v${RELEASE}.tar.gz")
-  tar -xzf v${RELEASE}.tar.gz
-  cp -r mafl-${RELEASE}/* /opt/mafl/
-  rm -rf mafl-${RELEASE}
-  cd /opt/mafl
-  yarn install
-  yarn build
-  systemctl start mafl
-  msg_ok "Updated Mafl to v${RELEASE}"
+  if [[ "${RELEASE}" != "$(cat ~/.mafl 2>/dev/null)" ]] || [[ ! -f ~/.mafl ]]; then
+    msg_info "Updating Mafl to v${RELEASE} (Patience)"
+    systemctl stop mafl
+    mkdir -p /opt/mafl-backup/data
+    mv /opt/mafl/data /opt/mafl-backup/data
+    rm /opt/mafl
+    fetch_and_deploy_gh_release "mafl" "hywax/mafl"
+    cd /opt/mafl
+    yarn install
+    yarn build
+    systemctl start mafl
+    mv /opt/mafl-backup/data /opt/mafl/data
+    msg_ok "Updated Mafl to v${RELEASE}"
+  else
+    msg_ok "No update required. ${APP} is already at v${RELEASE}"
+  fi
   exit
 }
 

--- a/install/mafl-install.sh
+++ b/install/mafl-install.sh
@@ -14,22 +14,17 @@ network_check
 update_os
 
 msg_info "Installing Dependencies"
-$STD apt-get install -y make
-$STD apt-get install -y g++
-$STD apt-get install -y gcc
-$STD apt-get install -y ca-certificates
+$STD apt-get install -y \
+  ca-certificates \
+  build-essential
 msg_ok "Installed Dependencies"
 
 NODE_VERSION="22" NODE_MODULE="yarn@latest" setup_nodejs
+fetch_and_deploy_gh_release "mafl" "hywax/mafl"
 
-RELEASE=$(curl -fsSL https://api.github.com/repos/hywax/mafl/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
 msg_info "Installing Mafl v${RELEASE}"
-curl -fsSL "https://github.com/hywax/mafl/archive/refs/tags/v${RELEASE}.tar.gz" -o "v${RELEASE}.tar.gz"
-tar -xzf v${RELEASE}.tar.gz
 mkdir -p /opt/mafl/data
 curl -fsSL "https://raw.githubusercontent.com/hywax/mafl/main/.example/config.yml" -o "/opt/mafl/data/config.yml"
-mv mafl-${RELEASE}/* /opt/mafl
-rm -rf mafl-${RELEASE}
 cd /opt/mafl
 export NUXT_TELEMETRY_DISABLED=true
 $STD yarn install

